core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLUtils.kt

```diff
@@ -165,4 +170,273 @@ object GraphQLUtils {
+                                addEdge(entryPoint, element.tableFieldType, element.tableField, graph)//from a node to the interface type object
+                                if (history.count { it == element.tableFieldType } <= 1) {// the interface type object already treated
+                                    constructGraph(state, element.tableFieldType, "", graph, history, objectFieldsHistory)//construct the base fields of the interface
+                                    element.interfaceTypes.forEach { ob ->
+                                        history.add(ob)//add the interface objects to the history
+                                        addEdge(element.tableFieldType, ob, interfaceTag, graph) // from the interface type object to its objects
+                                        if (history.count { it == ob } <= 1) {
+                                            constructGraph(state, ob, "", graph, history, objectFieldsHistory)
+                                            history.remove(ob)
+                                        } else {
+                                            history.remove(ob)
+                                        }
+                                    }
+                                    history.remove(element.tableFieldType)// remove the interface type object to the history:the normal case (not cycle)
+                                } else {
+                                    history.remove(element.tableFieldType)// remove the interface type object to the history:the cycle case
+                                }
+                            }
+                        }
+                    }
+
+                }
+            }
+        }
+        return graph
+    }
+
+    private fun addNode(node: String, graph: MutableMap<String, GraphInfo>): GraphInfo? {
+        return graph.putIfAbsent(node, GraphInfo())
+    }
+
+    private fun addEdge(source: String, destination: String, fieldName: String, graph: MutableMap<String, GraphInfo>): Boolean? {
+        addNode(source, graph)
+        addNode(destination, graph)
+        graph[source]?.edges?.add(destination)
+        return graph[source]?.fields?.add(fieldName)
+    }
+
+    /**
+     * The number of nodes
+     * Note: the count include Query and Mutation
+     */
+    fun getGraphSize(graph: MutableMap<String, GraphInfo>): Int {
+        return graph.size
+    }
+
+    /**
+     * The number of fields that are pointers to other nodes in the graph.
+     * Note: fields in Query and Mutation are considered in this count.
+     */
+    fun getNbrOfEdges(graph: MutableMap<String, GraphInfo>): Int {
+        return graph.values.stream().mapToInt { it.edges.size }.sum()
+    }
+
+    /**
+     * The number of fields in the type Query or mutation
+     */
+    fun getNbrQueriesOrMutations(queriesMutationsEntryPoint: String, graph: MutableMap<String, GraphInfo>): Int? {
```

Comment:
why returning a nullable Int? alsoe see comment on `Nbr` (it applies to many functions here).

Reply:
-Because of the function get:
```
kotlin.collections.Map public abstract operator fun get(
    key: K
): V?
Returns the value corresponding to the given key, or null if such a key is not present in the map.
```
-Done for nbr.